PDFReportEnergie/module.php

```diff
@@ -0,0 +1,250 @@
+        $object = $example->svg->g[0]->line;
+        for ($i = 0; $i < count($object); $i++) {
+            $node = $object[$i];
+
+            //set the opacity of the vertical lines on 0 (trancparence)
+            //"remove" the vertical lines
+            if (!empty($node) && $node['class'] == 'gridLine y') {
+                $node->addAttribute('opacity', '0');
+            }
+            if (!empty($node) && $node['class'] == 'gridLine y firstLine') {
+                $node->addAttribute('opacity', '0');
+            }
+
+            //set the horizontal lines grey
+            if (!empty($node) && $node['class'] == 'gridLine x' || $node['class'] == 'gridLine x firstLine') {
+                $node->addAttribute('opacity', '.2');
+            }
+        }
+
+        return $example->asXML();
+    }
+
+    private function GenerateHTMLText()
+    {
+        $data = $this->FetchData();
+
+        $title = strtoupper($this->Translate('Behave'));
+        $consum = sprintf($this->Translate('In %s you used up %s.'), $data[0], $data[1]);
+        $predictionText = $this->Translate('Expected usage based on your behavior:') . ' ' . $data[4];
+        $valueText = $this->Translate('Actual usage:') . ' ' . $data[1];
+        $consumText = $this->Translate("You could $data[6] your consumption by %s in the period");
+        $consumText2 = sprintf($consumText, $data[5]);
+
+        $text =
+        <<<EOT
+        <p> $consum </p>
+        </br></br>
+        <h2 style="font-weight: normal; font-size: 25px">$title</h2>
+        <hr style="height: 5px">
+        <br> </br>
+        <h3>$data[0]</h3>
+            <p>
+            $data[3] <br>
+            $predictionText <br>
+            $valueText <br/>
+            <br>
+            $consumText2<br/>
+            </p>
+        EOT;
+
+        if ($data[7] != -1) {
+            $text .= "<h3>CO2 Emmisionen</h3> <p>$data[7] kg</p>";
+        }
+        return $text;
+    }
+
+    private function FetchData()
+    {
+        $archivID = IPS_GetInstanceListByModuleID('{43192F0B-135B-4CE7-A0A7-1475603F3060}')[0];
+        $startTime = strtotime('first day of last month 00:00:00');
+        $endTime = strtotime('first day of this month 00:00:00');
+
+        $counterID = $this->ReadPropertyInteger('CounterID');
+
+        $month = $this->Translate(date('F', $startTime));
+
+        $consum = AC_GetAggregatedValues($archivID, $counterID, 3, $startTime, $endTime, 0)[0]['Avg'];
+        $consumLastYear = AC_GetAggregatedValues($archivID, $counterID, 3, strtotime('-1 year', $startTime), strtotime('-1 year', $endTime), 0)[0]['Avg'];
+
+        if (($temperatureID = $this->ReadPropertyInteger('TemperatureID')) != 0) {
+            $avgTemp = AC_GetAggregatedValues($archivID, $temperatureID, 3, $startTime, $endTime, 0)[0]['Avg'];
+            $avgTemp = round($avgTemp, 1);
+            $avgTemp = $this->Translate('Die Durchschnittstemperatur betrug: ') . GetValueFormattedEx($temperatureID, $avgTemp);
```

Comment:
Ich würde empfehlen Leerzeichen und Doppelpunkte (die in jeder Übersetzung drin sein müssen) aus dem zu übersetzenden String raus zu nehmen. Dadurch spart man sich Fehler beim Übersetzen.

Quasi: `$this->Translate('Die Durchschnittstemperatur betrug') . ': ' . ....`